src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp

```diff
@@ -268,10 +267,18 @@ class KerrSchild : public MarkAsAnalyticSolution {
       gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataType>,
       gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataType>>;
 
-  template <typename DataType>
-  tuples::tagged_tuple_from_typelist<tags<DataType>> variables(
-      const tnsr::I<DataType, volume_dim>& x, double t,
-      tags<DataType> /*meta*/) const noexcept;
+  template <typename DataType, typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataType, volume_dim>& x,
+                                         double /*t*/,
+                                         tmpl::list<Tags...> /*meta*/) const
+      noexcept {
+    static_assert(
+        tmpl2::flat_all_v<tmpl::list_contains_v<tags<DataType>, Tags>...>,
+        "At least one of the requested tags is not supported. The requested "
+        "tags are listed as template parameters of the `variables` function.");
+    IntermediateVars<DataType> intermediate(*this, x);
+    return {intermediate.get_var(Tags{})...};
```

Comment:
We noticed that the implementation with intermediate / cached variables is actually very similar to a databox with `Simple` and `ComputeTag`s. You could implement the precomputed variables as `SimpleTag`s and the cached or derived variables as `ComputeTag`s. Then `db::create` a databox and `db::get` the variables out of it in this function.
We think this is a tradeoff between code duplication and a perhaps optimized databox for this special case. What do you think?

Reply:
Given that the analytic solution is not critical for the performance of a simulation, perhaps we should prioritise readability and maintainability here.. then you can use the DataBox’s caching mechanism and the code in this class is reduced drastically. It could be a good example also for other solutions. What do you think?

Reply:
No, the solutions are critical for performance because they get called each time boundary conditions are applied. The reason this is done as one allocation is because of performance.

Another reason: the DataBox is not intended to be used outside of actions.

Reply:
Ok. Does the DataBox need more allocations because compute items are only allocated on demand, or what‘s the performance difference here? If so, would it make sense to make the DataBox compute item implementation more efficient?

Reply:
Yes, it requires more allocations. No, there isn't any simple (or maybe even reasonable) way to make it more efficient.

Reply:
I'm still on the fence about the caching mechanism that is being implemented for just this solution. This is something we also want other analytic solutions to do well if it indeed makes a measurable impact on performance (if not, why bother), but we should avoid re-implementing it in each solution. If you are set on avoiding a DataBox for this purpose, then how about we factor out as much as possible of the `IntermediateVars` code so it can be re-used by other solutions?